File: components/core/src/main/js/components/authentication/LoginForm.js
import { withRouter, Link } from 'react-router-dom';
import ListErrors from '../ListErrors';
import React from 'react';
import { inject, observer } from 'mobx-react';

var loginPanel = {
 padding: '5rem 6rem 4rem 6rem',
 minHeight: 'auto'
};

var forgotPassword = {
    color: '#fff'
}

@inject('authStore')
@withRouter
@observer
class LoginForm extends React.Component {

  componentWillUnmount() {
    this.props.authStore.reset();
  }

  handleUsernameChange = e => this.props.authStore.setUsername(e.target.value);

  handlePasswordChange = e => this.props.authStore.setPassword(e.target.value);

  handleSubmitForm = (e) => {
    e.preventDefault();
    this.props.authStore.login()
      .then(() => this.props.history.replace('/'));
  };

  render() {

    const { values, errors, inProgress } = this.props.authStore;


    return (

        <div className="container">

            <div className="col-md-2 hidden-sm hidden-xs">&nbsp;</div>

            <div className="col-md-8 col-xs-12">
                <h2>Log in to Search app</h2>

                <ListErrors errors={errors} />

                <form onSubmit={this.handleSubmitForm}>
                    <div className="panel panel-default" style={loginPanel} >
                        <div className="panel-body row">
                            <div className="col-lg-6">
                                <div className="form-group">
                                    <label>Username</label>

                                    <input
                                      className="form-control"
                                      type="text"
                                      placeholder="Username"
                                      value={values.username}
                                      onChange={this.handleUsernameChange}
                                      tabIndex="1"
                                    />
                                </div>
                            <div className="form-group">
                                <label>Password</label>

                                <input
                                  className="form-control"
                                  type="password"
                                  placeholder="Password"
                                  value={values.password}
                                  onChange={this.handlePasswordChange}
                                  tabIndex="1"
                                />
                            </div>
                        </div>
                    </div>

                    <button className="btn btn-primary btn-sm" id="updateProfile" tabIndex="3" type="submit">
                        &nbsp; LOG IN &nbsp;
                    </button>
                        <a className="" href="/password-recovery" style={forgotPassword} tabIndex="4">Forgot password?</a>
                    </div>
                </form>
            </div>

            <div className="col-md-2 hidden-sm hidden-xs">&nbsp;</div>

        </div>
    );
  }
}

export default LoginForm;

File: components/core/src/main/java/nextgen/templates/javascript/LoginForm.java
package nextgen.templates.javascript;

public class LoginForm {

	private final java.util.UUID uuid = java.util.UUID.randomUUID();
	private final org.stringtemplate.v4.STGroup stGroup;


	LoginForm(org.stringtemplate.v4.STGroup stGroup) {
		this.stGroup = stGroup;
	}

	@Deprecated
	public java.util.UUID uuid() {
		return uuid;
	}

	public java.util.UUID getUuid() {
		return uuid;
	}

	@Override
	public String toString() {
		final org.stringtemplate.v4.ST st = stGroup.getInstanceOf("LoginForm");
		return st.render().trim();
	}




	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		LoginForm that = (LoginForm) o;
		return uuid.equals(that.uuid);
	}

	@Override
	public int hashCode() {
		return java.util.Objects.hash(uuid);
	}

	static final String st = "LoginForm() ::= <<import { withRouter } from 'react-router-dom';\n" + 
				"import React from 'react';\n" + 
				"\n" + 
				"import { inject, observer } from 'mobx-react';\n" + 
				"\n" + 
				"import SignIn from '../components/SignIn.js';\n" + 
				"\n" + 
				"@withRouter\n" + 
				"@inject('authStore')\n" + 
				"@observer\n" + 
				"class LoginForm extends React.Component {\n" + 
				"\n" + 
				"	constructor(){\n" + 
				"		super();\n" + 
				"		this.handleSubmitForm = this.handleSubmitForm.bind(this);\n" + 
				"		this.handleUsernameChange = this.handleUsernameChange.bind(this);\n" + 
				"		this.handlePasswordChange = this.handlePasswordChange.bind(this);\n" + 
				"	}\n" + 
				"\n" + 
				"	handleSubmitForm = (e) => {\n" + 
				"		e.preventDefault();\n" + 
				"		this.props.authStore.login().then(() => this.props.history.replace('/')).catch(e => {});\n" + 
				"	}\n" + 
				"\n" + 
				"	handleUsernameChange = (e) => {\n" + 
				"		this.props.authStore.setUsername(e.target.value);\n" + 
				"	}\n" + 
				"\n" + 
				"	handlePasswordChange = (e) => {\n" + 
				"		this.props.authStore.setPassword(e.target.value);\n" + 
				"	}\n" + 
				"\n" + 
				"	render() {\n" + 
				"		return (<SignIn authStore={ this.props.authStore } onSubmit={ this.handleSubmitForm } />);\n" + 
				"	}\n" + 
				"}\n" + 
				"\n" + 
				"export default (LoginForm); >>";
}  